### ISSUE TITLE
Redirect bug in 4.4.0

### ISSUE THREAD

**Issue by ericwhrz** [square/okhttp#5845]
In version 4.4.0, I've noticed a bug where redirects do not work.
In an attempt to make a request to [http://yahoo.com/ads.txt](url), which redirects to [http://www.yahoo.com/ads.txt](url), okhttp 4.4.0 threw exception `javax.net.ssl.SSLPeerUnverifiedException: Failed to find a trusted cert that signed [...]`, where a lower version (e.g. 4.3.1) successfully fetches the contents from [http://www.yahoo.com/ads.txt](url).
Attaching the code to reproduce the bug:
```
import java.security.GeneralSecurityException;
import java.security.SecureRandom;
import java.security.cert.X509Certificate;
import java.util.Arrays;
import java.util.List;
import javax.net.ssl.SSLContext;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;

import okhttp3.ConnectionSpec;
import okhttp3.OkHttpClient;
import okhttp3.Protocol;
import okhttp3.Request;
import org.junit.Test;

public class TestClients {

    private final static X509TrustManager TRUST_MANAGER = new X509TrustManager() {
        @Override
        public void checkClientTrusted(X509Certificate[] x509Certificates, String s) {
        }

        @Override
        public void checkServerTrusted(X509Certificate[] x509Certificates, String s) {
        }

        @Override
        public X509Certificate[] getAcceptedIssuers() {
            return new X509Certificate[0];
        }
    };

    public static OkHttpClient createTrustingHttpClient() {
        try {
            final SSLContext sslContext = SSLContext.getInstance("TLS");
            sslContext.init(null, new TrustManager[] { TRUST_MANAGER }, new SecureRandom());

            return new OkHttpClient.Builder()
                    .sslSocketFactory(sslContext.getSocketFactory(), TRUST_MANAGER)
                    .connectionSpecs(Arrays.asList(ConnectionSpec.COMPATIBLE_TLS, ConnectionSpec.CLEARTEXT))
                    .protocols(List.of(Protocol.HTTP_1_1))
                    .build();
        } catch (GeneralSecurityException se) {
            throw new RuntimeException(se);
        }
    }

    @Test
    public void test() throws Exception {
        String url = "http://yahoo.com/ads.txt";

        OkHttpClient trustingHttpClient = createTrustingHttpClient();
        final var okrequest = new Request.Builder()
                .url(url)
                .build();
        final var okresponse = trustingHttpClient.newCall(okrequest).execute();
        System.out.println(okresponse.body().string());
    }
}
```

**Comment by yschimke** [comment]
It should be fixed in this PR https://github.com/square/okhttp/pull/5835

But it relates to connection coalescing failing badly when you disable all security.  I hope this is just in Dev?

**Comment by swankjesse** [comment]
Fixed in 4.4.1.
